### PR TITLE
fix(skeval/src): stop train() from mutating instance hyperparameters

### LINKED ISSUE #114: bug(skeval/src): train() permanently mutates instance hyperparameters as a side effect
`src/skeval/classifier/sentence_classifier.py` mutates `self.epochs`, `self.batch_size`, and `self.lr` when overrides are passed to `train()`.

```python
# current — permanently overwrites instance state
if epochs is not None:
    self.epochs = epochs
if batch_size is not None:
    self.batch_size = batch_size
if lr is not None:
    self.lr = lr
```

Calling `clf.train(sentences, labels, epochs=5)` on a classifier initialised with `epochs=10` permanently changes `clf.epochs` to `5`. Any subsequent call to `fit()` will silently use `epochs=5`. This contradicts how sklearn-compatible estimators are expected to behave - hyperparameters are set in `__init__` and not mutated at call time.

**Fix:** Use local variables inside `train()` instead of mutating `self`.

```python
_epochs = epochs if epochs is not None else self.epochs
_batch_size = batch_size if batch_size is not None else self.batch_size
_lr = lr if lr is not None else self.lr
```

### PR DESCRIPTION
**Pull request by Storm1289**
#### Reference Issues/PRs
Fixes #114

#### What does this implement/fix?
`train()` was permanently overwriting `self.epochs`, `self.batch_size`, and `self.lr` whenever call-time overrides were passed. The override silently leaked into every subsequent `fit()` call, breaking the sklearn estimator contract that hyperparameters set in `__init__` are not mutated at call time.

Fix: snapshot the three attributes before applying overrides, run `fit()` inside a `try`, and restore the snapshot in `finally` so the original state is preserved even if `fit()` raises.

#### Tests run

```bash
.venv/bin/pytest tests/ -v


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Bug Fixes**
  * Resolved an issue where hyperparameter overrides in the `train()` method would permanently modify instance settings. Hyperparameters now correctly revert to their original values after training, even if the process encounters an error.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

### CODE REVIEW

**Comment by Storm1289** [comment]
Hi @direkkakkar319-ops , please review when you have time : )

**Comment by direkkakkar319-ops** [comment]
@coderabbitai review

**Review by direkkakkar319-ops — APPROVED** [review]
Looks good !

File: tests/test_classifier.py [hunk 1]
```diff
@@ -293,22 +293,31 @@ def test_train_still_works():
...
     clf = SentenceClassifier(embed_dim=16, lr=0.005)
     with warnings.catch_warnings(record=True):
         warnings.simplefilter("always")
         clf.train(SENTENCES, LABELS, epochs=1, lr=0.123)
-    assert clf.lr == 0.123
+    assert clf.lr == 0.005
+
+
+def test_train_does_not_leak_overrides_into_fit():
+    """A train() override must not affect a subsequent fit() call."""
```
Comment (direkkakkar319-ops) [inline tests/test_classifier.py]:
```suggestion
    """train() override must not affect a subsequent fit() call."""
```

File: tests/test_classifier.py [hunk 2]
```diff
@@ -293,22 +293,31 @@ def test_train_still_works():
...
+    clf = SentenceClassifier(embed_dim=16, epochs=10)
+    with warnings.catch_warnings(record=True):
+        warnings.simplefilter("always")
+        clf.train(SENTENCES, LABELS, epochs=1)
+    assert clf.epochs == 10
```
Comment (direkkakkar319-ops) [inline tests/test_classifier.py:320]: we also need to add assertions for the batch_sizee and lr also they are missing in this test case

File: tests/test_classifier.py [hunk 3]
```diff
@@ -293,22 +293,31 @@ def test_train_still_works():
...
```
Comment (direkkakkar319-ops) [inline tests/test_classifier.py:321]: aslo add a seperate test case for epoches named this `def test_train_epochs_override_does_not_mutate():` the above will check for all and we will add each test case for each parameter 
this is for batch_siz - `test_train_batch_size_override_does_not_mutate`
this for lr - `test_train_lr_override_does_not_mutate`

so now you can add `test_train_epochs_override_does_not_mutate():` for epoches and modify the `est_train_does_not_leak_overrides_into_fit():` for all of the parameters

**Review by direkkakkar319-ops — CHANGES_REQUESTED** [review]

File: tests/test_classifier.py (hunk 2 above)
Comment (Storm1289) [reply tests/test_classifier.py:320]: okay :)

File: tests/test_classifier.py (hunk 3 above)
Comment (Storm1289) [reply tests/test_classifier.py:321]: Okay :)

**Review by direkkakkar319-ops — APPROVED** [review]
Thanks @Storm1289
